app/src/main/java/edu/uc/kovaciad/slicetracker/ui/main/MainViewModel.kt

```diff
@@ -10,7 +10,12 @@ import edu.uc.kovaciad.slicetracker.dto.Material
 import edu.uc.kovaciad.slicetracker.dto.Model
 import edu.uc.kovaciad.slicetracker.dto.Printer
 
-class MainViewModel : ViewModel() {
-    // TODO: Implement the ViewModel
+class MainViewModel {
+    //MutableLiveData array list
+    var material: MutableLiveData<Material> = MutableLiveData<Material>()
+    var brand: MutableLiveData<Brand> = MutableLiveData<Brand>()
 
+    fun fetchProduct(myMATERIAL: String) {
+
+    }
```

Comment:
ArrayList class is used to create a dynamic array in Kotlin. Dynamic array states that we can increase or decrease the size of an array as per requisites. It also provide read and write functionalities. ArrayList may contain duplicates and is non-synchronized in nature. We use ArrayList to access the index of the specified element, convert an ArrayList into string or another array and many more functionalities. Based on you program description I thought this would be helpful.
Check out : https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/-array-list/

Reply:
Are these used anywhere?  Also, fetchProduct(String) is empty; it's not fetching a product.  
This appears to be an incomplete feature.  In a code review, focus on reducing technical debt of existing code.  Avoid adding new, incomplete functionality.
1) Only features that meet the Definition of Done should be merged to Master.
2) Unused code is clutter, which increases technical debt.
This appears to be an incomplete feature.  In a code review, focus on reducing technical debt of existing code.  Avoid adding new, incomplete functionality.
1) Only features that meet the Definition of Done should be merged to Master.
2) Unused code is clutter, which increases technical debt.